.circleci/config.yml

```diff
@@ -179,27 +211,130 @@ jobs:
+            bash $PWD/scripts/publish_image 17.0.5 16.18.1 ${MBT_VERSION} "ghcr.io/sap"
+      - run:
+          name: publish Java 17.0.5 & Node 18.12.1 image
+          command: |
+            #Make sure HEAD points to master
+            git checkout master
+            git fetch
+            git rebase
+            MBT_VERSION=$(cat ./VERSION)
+            echo "Images release: ${MBT_VERSION}"
+            #Push to Docker Hub
+            echo "$DOCKER_HUB_TOKEN" | docker login --username $DOCKER_HUB_USER --password-stdin
+            bash $PWD/scripts/publish_image 17.0.5 18.12.1 ${MBT_VERSION} "devxci"
+            #Push to GitHub Container Registry
+            echo "$CLOUD_MTA_BOT_GITHUB_TOKEN" | docker login "ghcr.io" --username $CLOUD_MTA_BOT_USER --password-stdin
+            bash $PWD/scripts/publish_image 17.0.5 18.12.1 ${MBT_VERSION} "ghcr.io/sap"
+      - run:
+          name: publish Java 19.0.1 & Node 14.21.1 image
+          command: |
+            #Make sure HEAD points to master
+            git checkout master
+            git fetch
+            git rebase
+            MBT_VERSION=$(cat ./VERSION)
+            echo "Images release: ${MBT_VERSION}"
+            #Push to Docker Hub
+            echo "$DOCKER_HUB_TOKEN" | docker login --username $DOCKER_HUB_USER --password-stdin
+            bash $PWD/scripts/publish_image 19.0.1 14.21.1 ${MBT_VERSION} "devxci"
+            #Push to GitHub Container Registry
+            echo "$CLOUD_MTA_BOT_GITHUB_TOKEN" | docker login "ghcr.io" --username $CLOUD_MTA_BOT_USER --password-stdin
+            bash $PWD/scripts/publish_image 19.0.1 14.21.1 ${MBT_VERSION} "ghcr.io/sap"
+      - run:
+          name: publish Java 19.0.1 & Node 16.18.1 image
+          command: |
+            #Make sure HEAD points to master
+            git checkout master
+            git fetch
+            git rebase
+            MBT_VERSION=$(cat ./VERSION)
+            echo "Images release: ${MBT_VERSION}"
+            #Push to Docker Hub
+            echo "$DOCKER_HUB_TOKEN" | docker login --username $DOCKER_HUB_USER --password-stdin
+            bash $PWD/scripts/publish_image 19.0.1 16.18.1 ${MBT_VERSION} "devxci"
+            #Push to GitHub Container Registry
+            echo "$CLOUD_MTA_BOT_GITHUB_TOKEN" | docker login "ghcr.io" --username $CLOUD_MTA_BOT_USER --password-stdin
+            bash $PWD/scripts/publish_image 19.0.1 16.18.1 ${MBT_VERSION} "ghcr.io/sap"
+      - run:
+          name: publish Java 19.0.1 & Node 18.12.1 image
+          command: |
+            #Make sure HEAD points to master
+            git checkout master
+            git fetch
+            git rebase
+            MBT_VERSION=$(cat ./VERSION)
+            echo "Images release: ${MBT_VERSION}"
+            #Push to Docker Hub
+            echo "$DOCKER_HUB_TOKEN" | docker login --username $DOCKER_HUB_USER --password-stdin
+            bash $PWD/scripts/publish_image 19.0.1 18.12.1 ${MBT_VERSION} "devxci"
```

Comment:
You have duplicated the same code needlessly increasing: 
+ the version maintenance overhead
+ the number of call to the docker hub login 
+ the number of duplicate same tasks at image build and push

Reply:
The problem is, if we put all build process for docker images in one "workflow node" of circleCI, it will use a longtime to run, the circleCI workflow node will run expire and failed.

Reply:
Fair enough, but you can probably factor out that code into a shared job required by all the others with circleCI?

Reply:
No, as I know about circleCI, it's schema is not to support only define one common workflow-node with parameter of java and node version, and workflow use it to define multiple different workflow-node instance. 

Maybe I can investigate circleCI deeper in the future, but for this commit, I just duplicate the code for different docker image.